.github/workflows/knative-releasability.yaml

```diff
@@ -40,35 +42,65 @@ jobs:
           go-version: 1.15.x
 
       - name: Install Dependencies
-        run: |
-          GO111MODULE=on go get tableflip.dev/buoy@master
+        run: GO111MODULE=on go get tableflip.dev/buoy@master
 
       - name: Check out code
         uses: actions/checkout@v2
+        with:
+          path: ./src/knative.dev/${{ github.event.repository.name }}
 
       - name: Check
         run: |
-          buoy check go.mod --release $RELEASE --verbose
+          echo 'CHECK_MESSAGE<<EOF' >> $GITHUB_ENV
+          (buoy check go.mod --release ${RELEASE} --domain knative.dev --verbose || CHECK=$1) >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+
+          if [[ "$CHECK" -eq "0"  ]]; then
+            echo 'current=true' >> $GITHUB_ENV
+          else
+            echo 'current=false' >> $GITHUB_ENV
+          fi
+
+      - name: Upgrade
+        if: env.current == 'true'
+        run: |
+          if ./hack/update-deps.sh --release ${RELEASE} --upgrade; then
```

Comment:
What is this doing?

Reply:
this is bash `if` testing the return code, so if the hack command returns 0 (success), bash does the `then`

Reply:
Added some comments to help